### PR TITLE
#86 - read project metadata on upload

### LINKED ISSUE #86: Read project metadata file on upload
Python project name should be obtained from project archive on upload by unpacking the archive and reading project metadata file. Package format is described by [PEP 427](https://www.python.org/dev/peps/pep-0427/)

### PR DESCRIPTION
**Pull request by olenagerasimova**
Closes #86.
Read python project name and use it as storage key to save artifact. Note that a bit later we will verify artifact name correspondence with metadata information.

### CODE REVIEW

File: src/main/java/com/artipie/pypi/http/WheelSlice.java [hunk 1]
```diff
@@ -68,24 +74,47 @@ public Response response(
...
-                    new Key.From(
-                        new KeyFromPath(new RequestLineFrom(line).uri().toString()),
-                        new NormalizedProjectName.FromFilename(data.fileName()).value(),
-                        data.fileName()
-                    ),
-                    new Content.From(data.bytes())
-                )
+            new Multipart(iterable, publisher).content().thenCompose(
+                data -> {
+                    final Path path = WheelSlice.save(data.bytes());
```
Comment (olegmoz) [inline src/main/java/com/artipie/pypi/http/WheelSlice.java:79]: @olenagerasimova I'd suggest to use `storage` to store temporary file. You can save data by random key first and the use `move` method to rename it. It is going to work faster and will avoid unexpected load on file system.

Comment (olenagerasimova) [reply src/main/java/com/artipie/pypi/http/WheelSlice.java:79]: @olegmoz thanks, I'm thinking to rewrite this saving to temp file part, now it looks not optimized: bytes are being hold into memory and saved to file at the same time.

**Comment by olenagerasimova** [comment]
@olegmoz thanks, corrected, please, take a look

**Comment by 0crat** [comment]
Job `gh:artipie/pypi-adapter#98` is not assigned, can't get performer

**Pull request merged**